webui/react/src/omnibar/Omnibar.tsx

```diff
@@ -1,37 +1,59 @@
 import OmnibarNpm from 'omnibar';
-import React, { useCallback } from 'react';
+import React, { useCallback, useEffect } from 'react';
 
-import { StoreAction, useStoreDispatch } from 'contexts/Store';
+import { StoreAction, useStore, useStoreDispatch } from 'contexts/Store';
 import * as Tree from 'omnibar/tree-extension/index';
 import TreeNode from 'omnibar/tree-extension/TreeNode';
 import { BaseNode } from 'omnibar/tree-extension/types';
 import { isTreeNode } from 'omnibar/tree-extension/utils';
+import handleError from 'utils/error';
 
 import css from './Omnibar.module.scss';
 
+/*
+Ideally we wouldn't need to access the element like this.
+A potential option is use the value prop in combinatio with encoding the tree path into
+the options returned by the tree extension.
+*/
+const omnibarInput = () => document.querySelector(
+  '#omnibar input[type="text"]',
+) as (HTMLInputElement | null);
+
 const Omnibar: React.FC = () => {
   const storeDispatch = useStoreDispatch();
+  const { ui } = useStore();
 
-  const hideBar = useCallback(
-    () => storeDispatch({ type: StoreAction.HideOmnibar }),
-    [ storeDispatch ],
-  );
+  const hideBar = useCallback(() => {
+    storeDispatch({ type: StoreAction.HideOmnibar });
+  }, [ storeDispatch ]);
+
+  const onAction = useCallback(async (item, query) => {
+    const input: HTMLInputElement|null = omnibarInput();
 
-  const onAction = useCallback((item, query) => {
-    /*
-    Ideally we wouldn't need to access the element like this.
-    A potential option is use the value prop in combinatio with encoding the tree path into
-    the options returned by the tree extension.
-    */
-    const input: HTMLInputElement|null = document.querySelector('#omnibar input[type="text"]');
     if (!input) return;
     if (isTreeNode(item)) {
-      return Tree.onAction(input, item, query);
+      try {
+        await Tree.onAction(input, item, query);
+        if (item.closeBar) {
+          hideBar();
+        }
+      } catch (e) {
+        handleError(e);
+      }
+    }
+  }, [ hideBar ]);
+
+  useEffect(() => {
+    const input: HTMLInputElement|null = omnibarInput();
+    if (ui.omnibar.isShowing) {
+      if (input) {
+        input.focus();
```

Comment:
**suggestion:** add `input.select();` below the focus? With the preservation of the previous commands, I find that majority of the time I want to do a different command from the previous so having all of it highlighted by default saves me the trouble of having to do a `ctrl/cmd + A` to highlight everything, then hit delete or enter a new command.